src/App.tsx

```diff
@@ -15,35 +16,105 @@ export const goodsFromServer = [
+  Empty = '',
+}
+
+interface SortParams {
+  list: string[];
+  sortBy: SortType | '';
+  reverse: boolean;
+}
+
+const sortGoods = ({ list, sortBy, reverse }: SortParams) => {
+  const preparedList = [...list];
+
+  if (sortBy) {
+    switch (sortBy) {
+      case SortType.Alphabetic:
+        preparedList.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+
+      case SortType.Length:
+        preparedList.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (reverse) {
+    preparedList.reverse();
+  }
+
+  return preparedList;
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortLabel, setSortLabel] = useState(SortType.Empty);
+  const isResetable = sortLabel !== SortType.Empty || isReversed !== false;
+
+  const sortedList = sortGoods({
+    list: goodsFromServer,
+    sortBy: sortLabel,
+    reverse: isReversed,
+  });
+
+  const resetGoods = () => {
+    setIsReversed(false);
+    setSortLabel(SortType.Empty);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          type="button"
+          onClick={() => setSortLabel(SortType.Alphabetic)}
+          className={classNames('button is-info', {
+            'is-light': sortLabel !== SortType.Alphabetic,
+          })}
+        >
+          Sort Alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': sortLabel !== SortType.Length,
+          })}
+          onClick={() => setSortLabel(SortType.Length)}
+        >
+          Sort by Length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isResetable && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => resetGoods()}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {sortedList && sortedList.map(good => <li data-cy="Good">{good}</li>)}
```

Comment:
Don't forget about the key attribute
```suggestion
          {sortedList.map(good => <li data-cy="Good">{good}</li>)}
```